### ISSUE TITLE
[Bug] Reader view tabs randomly revert to default font/color/size, when switching apps

### ISSUE THREAD

**Issue by nyanpasu64** [mozilla-mobile/fenix#18792]
[comment]: # (Please do your best to search for duplicate issues before filing a new issue so we can keep our issue board clean)
[comment]: # (Every issue should have the exact bug and steps to reproduce described in it. Please do not file feedback list tickets as it is difficult to parse them and address their individual points)
[comment]: # (Read https://github.com/mozilla-mobile/fenix#i-want-to-file-an-issue for more information)

## Steps to reproduce

- Open "Developer options" and enable "Don't keep activities".
- Open Firefox to an article, and enable reader view.
- In the reader view options, set the font family to sans-serif, change the size, and/or switch color schemes.
- Switch to a different Firefox tab.
- Switch to a different recent app. Wait a second or so, then switch back to Firefox.
- Switch back to the reader view tab.

### Expected behavior
The article keeps the font family, size, and color scheme you have selected.

### Actual behavior
The article redraws in the default font family (serif), size, and color scheme (light). Switching the font to serif and back fixes the issue (tapping sans-serif doesn't help). Switching to any other color scheme fixes the issue. Changing the font size doesn't fix the issue directly, until you shrink the size to minimum or maximum (causing the appearance to stop changing while the button presses still occur).

Reloading the page fixes the issue and reverts to the appearance you've picked.

### Device information

* Device vendor / model and Android version: moto g power (2021), Android 10
* Firefox for Android version: Nightly 210403 17:02 (Build #2015802507); AC: 75.0.20210403143100, 9bd53c36f; GV: 89.0a1-20210403093157; AS: 74.0.1

**Comment by ebalazs-sv** [comment]
I can reproduce this issue on Nightly 4/6, Beta 88.0.0-beta.2 and RC 87.0.0-rc.1 with Pixel 2 (Android 9).
![20210406_161232](https://user-images.githubusercontent.com/48799384/113716480-50198080-96f3-11eb-88a7-a283dfce3207.gif)

**Comment by nyanpasu64** [comment]
I just got this issue again without "Don't keep activities" checked (I initially discovered the big randomly without the thing checked, I only use it as a way to reliably replicate the bug). But this time it affected the tab I had open since I didn't switch away from the reader tab, and the theme reset a split-second after I switched back to Firefox (when Firefox switched from a static screenshot to rerendering the page). I suppose it's the same underlying bug.

Forgot to mention, once the bug triggers in any one tab, all open reader view tabs have their theme reset simultaneously.
